Assets/Tests/InputSystem/Plugins/UITests.cs

```diff
@@ -4016,6 +4016,260 @@ public IEnumerator UI_DisplayIndexMatchesDisplayMultiplePointers()
+    public Vector2 MousePosition = Vector2.zero;
+    public Vector2 MouseScrollDelta = Vector2.zero;
+    public bool TouchSupported = false;
+    public int TouchCount = 0;
+    public Touch TouchData;
+    public float AxisRaw = 0f;
+    public bool ButtonDown = false;
+
+    public Image image;
+
+    void Update()
+    {
+        MousePosition.x += 1f;
+        image.GetComponent<RectTransform>().position = MousePosition;
+    }
+
+    public override string compositionString
+    {
+        get { return CompositionString; }
+    }
+
+    public override IMECompositionMode imeCompositionMode
+    {
+        get { return m_ImeCompositionMode; }
+        set { m_ImeCompositionMode = value; }
+    }
+
+    public override Vector2 compositionCursorPos
+    {
+        get { return m_CompositionCursorPos; }
+        set { m_CompositionCursorPos = value; }
+    }
+
+    public override bool mousePresent
+    {
+        get { return MousePresent; }
+    }
+
+    public override bool GetMouseButtonDown(int button)
+    {
+        return MouseButtonDown[button];
+    }
+
+    public override bool GetMouseButtonUp(int button)
+    {
+        return MouseButtonUp[button];
+    }
+
+    public override bool GetMouseButton(int button)
+    {
+        return MouseButton[button];
+    }
+
+    public override Vector2 mousePosition
+    {
+        get { return MousePosition; }
+    }
+
+    public override Vector2 mouseScrollDelta
+    {
+        get { return MouseScrollDelta; }
+    }
+
+    public override bool touchSupported
+    {
+        get { return TouchSupported; }
+    }
+
+    public override int touchCount
+    {
+        get { return TouchCount; }
+    }
+
+    public override Touch GetTouch(int index)
+    {
+        return TouchData;
+    }
+
+    public override float GetAxisRaw(string axisName)
+    {
+        return AxisRaw;
+    }
+
+    public override bool GetButtonDown(string buttonName)
+    {
+        return ButtonDown;
+    }
+}
+
+public class DragCallbackCheck : MonoBehaviour, IBeginDragHandler, IDragHandler, IEndDragHandler, IDropHandler, IPointerDownHandler
+{
+    private bool loggedOnDrag = false;
+    public bool onBeginDragCalled = false;
+    public bool onDragCalled = false;
+    public bool onEndDragCalled = false;
+    public bool onDropCalled = false;
+
+    public void OnBeginDrag(PointerEventData eventData)
+    {
+        Debug.Log(1);
+        onBeginDragCalled = true;
+    }
+
+    public void OnDrag(PointerEventData eventData)
+    {
+        if (loggedOnDrag)
+            return;
+
+        loggedOnDrag = true;
+        onDragCalled = true;
+        Debug.Log(2);
+    }
+
+    public void OnEndDrag(PointerEventData eventData)
+    {
+        Debug.Log(3);
+        onEndDragCalled = true;
+    }
+
+    public void OnDrop(PointerEventData eventData)
+    {
+        Debug.Log(4);
+        onDropCalled = true;
+    }
+
+    public void OnPointerDown(PointerEventData eventData)
+    {
+        Debug.Log(5);
+    }
+}
```

Comment:
The test harness will take care of all event capturing for you. You can access it inside the `scene` object.